src/main/webapp/script.js

```diff
@@ -10,11 +15,49 @@ function passQuery() {
         .then((data) => { 
           // data is a 2D array, where the first row is a header row and all
           // subsequent rows are one piece of data (e.g. for a state or county)
-          callOliviaFunction(data);
+          displayVisualization(data);
         });
       }
       else {
         console.log("There was an error");
       }
     });
-}
\ No newline at end of file
+}
+
+ // displayVisualization takes in a data array representing the 2D array returned by the 
+ // census API. The function creates the necessary visualization with the census data.
+function displayVisualization(censusDataArray) {
+  if (censusDataArray[0][2] == 'state' && censusDataArray[0][3] != 'county') {
+    google.charts.setOnLoadCallback(drawRegionsMap(censusDataArray));
+  } else {
+    // We currently do not have counties implemented
+    document.getElementById('map').innerHTML = "";
+    document.getElementById('more-info').innerHTML = 'We do not have this information yet.'
+  }
+}
+
+// Takes in a 2D array from the census API and draws the appropriate visualization
+function drawRegionsMap(censusDataArray) {
+  const shortDataArray = createDataArray(censusDataArray);
+  var data = google.visualization.arrayToDataTable(shortDataArray);
+  var options = {
+    'region': 'US',
+    'resolution': 'provinces',
+    'colorAxis': {colors: ['white', 'blue']}
+  };
+  var chart = new google.visualization.GeoChart(document.getElementById('map'));
+  chart.draw(data, options);
+  document.getElementById('more-info').innerHTML = 'These populations are divided by 1000';
+}
+
+// createDataArray takes in the data array returned by the census API and reformats it 
+// into a data table that can be processed by the visualization API.
+function createDataArray(censusDataArray) {
+  const vizDataArray = [];
+  censusDataArray.forEach((state) => {
+    vizDataArray.push([state[0], state[1]/1000]);
+  });
+  vizDataArray[0][0] = 'State';
```

Comment:
Would this not replace the first value you pushed? How about calling vizDataArray.push(['State', 'Population']) before the foreach loop.

Reply:
I actually wrote this line of code, so I can explain. The data that comes from the census has its own header row, but the google charts visualization doesn't recognize its terms. So this replaces that header row with one that the charts api can understand. There might be a better way to write that, though.

Reply:
This definitely warrant a comment explaining this in the code.

Reply:
fixed